src/bin/nydusd/fs_cache.rs

```diff
@@ -462,17 +462,21 @@ impl FsCacheHandler {
             e.insert((FsCacheObject::DataBlob(fsblob.clone()), msg.fd));
             state.id_to_config_map.insert(hdr.object_id, config.clone());
             let blob_size = config.blob_info().deref().uncompressed_size();
-            Self::init_blobcache(fsblob);
+            let barrier = Arc::new(Barrier::new(2));
+            Self::init_blobcache(fsblob, Arc::clone(&barrier));
+            // make sure blob cache init thread get wlock before get first ondemand req
+            barrier.wait();
```

Comment:
Nit: 
Some grammar mistakes  with this comment sentence 😹

Reply:
fixed ~